torch/csrc/utils/python_arg_parser.cpp

```diff
@@ -107,19 +105,27 @@ std::string FunctionParameter::type_name() const {
 }
 
 void FunctionParameter::set_default_str(const std::string& str) {
+  if (str == "None") {
+    allow_none = true;
+  }
   if (type_ == ParameterType::TENSOR) {
     if (str != "None") {
       throw std::runtime_error("default value for Tensor must be none, got: " + str);
     }
-    return;
   } else if (type_ == ParameterType::INT64) {
     default_int = atol(str.c_str());
   } else if (type_ == ParameterType::BOOL) {
     default_bool = (str == "True" || str == "true");
   } else if (type_ == ParameterType::DOUBLE) {
     default_double = atof(str.c_str());
   } else if (type_ == ParameterType::SCALAR) {
-    default_scalar = Scalar(atof(str.c_str()));
+    if (str == "None") {
+      // This is a bit awkward, but convenient for clamp which takes Scalars,
+      // but allows None.
+      default_scalar = Scalar(NAN);
```

Comment:
I should never actually read out this value, right? (because of how the dispatch works for clamp).  Do you envision we will use this?  If not, should we just throw an exception at the read site (i.e. make the read a function), because the right behavior is to throw an error before dispatch, not to dispatch to the function with NaNs.

Reply:
That would be better, but I don't see a way to do it without complicating the code.